src/lib/classes/services/chainsService.ts

```diff
@@ -131,4 +125,15 @@ export class ChainsService {
     public save() {
         this._storageService.save("chains", this._cachedChains);
     }
+
+    private cacheChainBlock(chainID: string, block: BlockData) {
+        const MAX_CACHED_BLOCKS = 20;
+        if (this._cachedChains[chainID].blocks.length >= MAX_CACHED_BLOCKS) {
+            this._cachedChains[chainID].blocks.shift();
+        }
+        this._cachedChains[chainID].blocks = this._cachedChains[chainID].blocks
```

Comment:
Also, wouldnt this code potentially cache the same block N times? We are lacking a check that the block doesnt exist already

Reply:
Indeed its caching the same block N times 
![image](https://github.com/iotaledger/wasp-dashboard/assets/3624944/a692c02f-b612-45c7-91eb-c4d83ee7d37a)